lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1269,8 +1270,13 @@ Stmt *StmtChecker::visitBraceStmt(BraceStmt *BS) {
       break;
 
     TC.typeCheckDecl(SubDecl, /*isFirstPass*/false);
+    if (SubDecl->isInvalid())
```

Comment:
I'm not a fan of introducing new usages of the 'invalid' bit.

Also consider a case where SubDecl is not invalid, but one of it's child decls is (and maybe that child decl has a child with a null type somewhere...)

I think declaration checking should leave things in a state where everything has a type, at least. What do you think?